src/app/account-sharing/home/[username]/AccountSharingHome.tsx

```diff
@@ -123,21 +116,25 @@ export function AccountSharingHome({ username, embed }: { username: string; embe
      * In a real-world application, you might opt to use a server-sent events (SSE) or web sockets to get real-time updates.
      */
     refetchInterval: 2000,
-    onSuccess: (data) => {
-      if (data.severity === 'error') {
-        youHaveBeenLoggedOut(
-          data.otherDevice ? `${data.otherDevice.deviceName} (${data.otherDevice.deviceLocation})` : undefined,
-        );
-      }
-    },
-    onSettled: () => {
-      setIsInitialLoading(false);
-    },
   });
 
+  // To display the loading state even before Fingerprint JavaScript agent is loaded
+  const isInitialLoading = !loggedInData && !loggedInError;
+
+  // Handle side-effects previously in onSuccess and onSettled
+  useEffect(() => {
+    if (loggedInData?.severity === 'error') {
+      youHaveBeenLoggedOut(
+        loggedInData.otherDevice
+          ? `${loggedInData.otherDevice.deviceName} (${loggedInData.otherDevice.deviceLocation})`
+          : undefined,
+      );
+    }
+  }, [loggedInData, loggedInError, youHaveBeenLoggedOut]);
```

Comment:
`loggedInError` is not used in this hook, do we need to keep it in the deps array?